File: app/src/main/java/info/maaskant/wouttest2/detail/ViewerFragment.java
package info.maaskant.wouttest2.detail;

import javax.inject.Inject;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import info.maaskant.wouttest2.R;
import info.maaskant.wouttest2.WoutTest2App;
import info.maaskant.wouttest2.utils.ApplicationInstrumentation;
import timber.log.Timber;

public class ViewerFragment extends Fragment {

    @Inject
    DetailViewModel detailViewModel;

    @Inject
    ApplicationInstrumentation instrumentation;

    private ViewerView.ViewBinder viewerViewBinder;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((WoutTest2App) getActivity().getApplication()).getGraph().inject(this);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.viewer_fragment, container, false);
        return view;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        viewerViewBinder = new ViewerView.ViewBinder(
                (ViewerView) view.findViewById(R.id.viewer_fragment), detailViewModel);
        detailViewModel.subscribeToDataStore();
    }

    @Override
    public void onStart() {
        super.onStart();
        Timber.v("onStart");
    }

    @Override
    public void onResume() {
        super.onResume();
        Timber.v("onResume");

        viewerViewBinder.bind();
    }

    @Override
    public void onPause() {
        super.onPause();
        Timber.v("onPause");
        viewerViewBinder.unbind();
    }

    @Override
    public void onStop() {
        super.onStop();
        Timber.v("onStop");
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        Timber.v("onDestroyView");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Timber.v("onDestroy");
        detailViewModel.dispose();
        instrumentation.getLeakTracing().traceLeakage(this);
    }

}

File: app/src/main/java/info/maaskant/wmsnotes/android/ui/detail/ViewerFragment.kt
package info.maaskant.wmsnotes.android.ui.detail

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import dagger.hilt.android.AndroidEntryPoint
import info.maaskant.wmsnotes.R
import info.maaskant.wmsnotes.utilities.logger
import javax.inject.Inject

@AndroidEntryPoint
class ViewerFragment @Inject constructor() : Fragment() {
    private val logger by logger()

    private val detailViewModel: DetailViewModel by activityViewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        logger.trace("onCreate (this: {})", System.identityHashCode(this))
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        logger.trace(
            "onCreateView (this: {}, vm: {})",
            System.identityHashCode(this),
            System.identityHashCode(detailViewModel)
        )
        return inflater.inflate(R.layout.viewer_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        logger.trace("onViewCreated (this: {})", System.identityHashCode(this))
        super.onViewCreated(view, savedInstanceState)
        val textView = view.findViewById<TextView>(R.id.viewer_textview)
        lifecycle.addObserver(Renderer(detailViewModel, requireContext(), textView))
    }
}

